docs/architecture/decisions/0006-garden-linux-versioning.md

```diff
@@ -0,0 +1,55 @@
+# 6. Garden Linux versioning
+
+Date: 2025-08-25
+
+## Status
+
+Accepted
+
+## Context
+
+The Gardenlinux version scheme is based on a 2 segment version identifier scheme under the format DDDD.P where:
+- DDDD represents the number of days since 31st of March 2020
+- P minor version
+
+The format is not extremely strict - no padding is being used e.g. 0934.01 doesn't exist, rather 934.1 was released.
+The Gardenlinux version identifier is very similar to what the CalVer[^1] convention is proposing, except the 3 numeric 
+segments and the modifier.
+
+Because of different limitations from our main customer (Gardener) and as most OSS is using SemVer[^2] as the go-to 
+versioning system, it was proposed that Gardenlinux should also switch to SemVer.
+
+## Decision
+
+Unfortunately SemVer is not really suitable for a such a project.
+
+In order to align with the requests, the decision has been made to:
+- keep the current system we use for major/minor - similar to what CalVer is proposing, even if you can't easily derive a lot of information from the version itself
+- switch to a 3 numerical segments identifier for the version:
+  - major - keep the same convention - number of days since 31st of March 2020
+  - minor
+  - patch - always 0 for GL - forks of GL can increment this in order to release their own versions based on our major.minor.0 releases
```

Comment:
Just as an alternative thought. In my opinion GardenLinux can shift to an Semantic Versioning Scheme taking the approach of Debian, Python and other projects (like we do for packages ourself) to add a suffix for third party forks.

`<major>.<minor>.<patch>` - would therefore mean:
1. Development freezes result in new major releases
2. Subsequent releases of an updated major release either bump the "patch" digit if no component received an intended version bump (e.g. rare update in `runc`, `containerd` or `golang`) or a "minor" digit bump if an intended component version bump has been added.
3. Third-party forks can easily determine which GL version they use in leaving the semantic versioning from GardenLinux and adding something like `-alpha1`, `-post1` or `-my-cool-fork-1.2` to it.